### PR TITLE
statistics: add a test case which builds global-stats on different versions of partition-stats

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: add a test case which builds global-stats on different versions of partition-stats

### What is changed and how it works?

statistics: add a test case which builds global-stats on different versions of partition-stats

### Check List 

Tests 

- Unit test

### Release note 

- statistics: add a test case which builds global-stats on different versions of partition-stats

### CODE REVIEW

**Review by rebelice — APPROVED** [review]

File: statistics/handle/handle_test.go [hunk 1]
```diff
@@ -1266,18 +1266,24 @@ partition by range (a) (
...
 
+	tk.MustExec("set @@tidb_partition_prune_mode='dynamic'")
 	tk.MustExec("set @@tidb_analyze_version=2")
-	tk.MustExec("analyze table t")
-	c.Assert(len(tk.MustQuery("show stats_meta").Rows()), Equals, 3) // p0 + p1 + global
+	err = tk.ExecToErr("analyze table t partition p1") // only analyze p1 to let it in ver2 while p0 is in ver1
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[stats]: some partition level statistics are not in statistics version 2, please set tidb_analyze_version to 2 and analyze the this table")
+
+	tk.MustExec("analyze table t") // both p0 and p1 are in ver2
```
Comment (Reminiscent) [inline statistics/handle/handle_test.go:1285]: Please add some tests to prove that when partition `p1` contains ver2-stats, we can use `analyze table t partition p0` to get the global-stats.

**Review by Reminiscent — COMMENTED** [review]
REST LGTM

File: statistics/handle/handle_test.go (hunk 1 above)
Comment (qw4990) [reply statistics/handle/handle_test.go:1345]: OK, more cases have been added.

**Review by Reminiscent — APPROVED** [review]
/LGTM

**Pull request merged**